AppCenterAnalytics/AppCenterAnalytics/Internals/Model/Properties/MSBooleanTypedProperty.m

```diff
@@ -0,0 +1,49 @@
+#import "MSBooleanTypedProperty.h"
+#import "MSConstants+Internal.h"
+
+@implementation MSBooleanTypedProperty
+
+- (instancetype)init {
+  if ((self = [super init])) {
+    self.type = @"boolean";
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder {
+  self = [super initWithCoder:coder];
+  if (self) {
+    _value = [coder decodeBoolForKey:kMSTypedPropertyValue];
+  }
+  return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)coder {
+  [super encodeWithCoder:coder];
+  [coder encodeBool:self.value forKey:kMSTypedPropertyValue];
+}
+
+/**
+ * Serialize this object to a dictionary.
+ *
+ * @return A dictionary representing this object.
+ */
+- (NSMutableDictionary *)serializeToDictionary {
+  NSMutableDictionary *dict = [super serializeToDictionary];
+  dict[kMSTypedPropertyValue] = @(self.value);
+  return dict;
+}
+
+- (instancetype)createValidCopyForAppCenter {
```

Comment:
Can you decouple business logic from data model?

Reply:
For all typed property classes.

Reply:
`MSAnalytics+Validation` would be the right place.

Reply:
I agree, it's better if the validation happens in only one place.

Reply:
Sure, but since we only will have an `MSTypedProperty` we will need to switch on type and then cast to whatever it is to validate, which is quite unfortunate. (That's why we decided to move it into the data models themselves.)